packages/types/src/codec/Result.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017-2019 @polkadot/types authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { Codec, Constructor, InterfaceTypes } from '../types';
+
+import { assert } from '@polkadot/util';
+
+import Enum from './Enum';
+
+/**
+ * @name Result
+ * @description
+ * A Result maps to the Rust Result type, that can either wrap a success or error value
+ */
+export default class Result<O extends Codec, E extends Codec> extends Enum {
+  public constructor (Ok: Constructor | InterfaceTypes, Error: Constructor | InterfaceTypes, value?: any) {
```

Comment:
one small request: can we do `constructor (def: {Ok: ..., Error: ...}, value?: any)` here, instead of 3 args, to keep similar signature to Enum?

Reply:
That would mean it is order dependent and at some point somebody is going to "remove the useless" constructor. Additionally, `Result<O, E>` does actually take 2 params, not a single.

The `with` taking in `{ Ok: ..., Error: ... }` is already problematic here. (Which is required since it extends `Enum`, normally you would go `{ Error: ..., Ok: ... }` since that . is alpabetical)